pkg/overlay/cache.go

```diff
@@ -92,6 +98,74 @@ func (cache *Cache) Get(ctx context.Context, nodeID storj.NodeID) (*pb.Node, err
 	return cache.db.Get(ctx, nodeID)
 }
 
+// FindStorageNodes searches the overlay network for nodes that meet the provided criteria
+func (cache *Cache) FindStorageNodes(ctx context.Context, req *pb.FindStorageNodesRequest, preferences *NodeSelectionConfig) ([]*pb.Node, error) {
+	// TODO: use a nicer struct for input
```

Comment:
what would be a nicer struct for input? where you don't have to dig through `GetOpts().GetRestrictions()`?